tests/countries/test_thailand.py

```diff
@@ -561,33 +601,167 @@ def test_khao_phansa(self):
-            "2019-05-09",
-            "2020-05-11",
-            "2021-05-13",
-            "2022-05-17",
-            "2023-05-11",
+        self.assertHolidays(
+            Thailand(categories=(GOVERNMENT,), years=range(1997, 2024)),
+            ("1997-05-13", name),
+            ("1998-05-13", name),
+            ("2000-05-15", name),
+            ("2001-05-16", name),
+            ("2002-05-09", name),
+            ("2003-05-08", name),
+            ("2004-05-07", name),
+            ("2005-05-11", name),
+            ("2006-05-11", name),
+            ("2007-05-10", name),
+            ("2008-05-09", name),
+            ("2009-05-11", name),
+            ("2010-05-10", name),
+            ("2011-05-13", name),
+            ("2012-05-09", name),
+            ("2013-05-13", name),
+            ("2014-05-09", name),
+            ("2015-05-13", name),
+            ("2016-05-09", name),
+            ("2017-05-12", name),
+            ("2018-05-14", name),
+            ("2019-05-09", name),
+            ("2020-05-11", name),
+            ("2021-05-13", name),
+            ("2022-05-17", name),
+            ("2023-05-11", name),
+        )
+
+    def test_armed_forces_holiday(self):
+        name = "วันกองทัพไทย"
+        self.assertHolidays(
+            Thailand(categories=(ARMED_FORCES,), years=range(1958, 1960)), ("1959-04-08", name)
+        )
+        self.assertHolidays(
+            Thailand(categories=(ARMED_FORCES,), years=range(1979, 1981)),
+            ("1979-04-08", name),
+            ("1980-01-25", name),
+        )
+        self.assertHolidays(
+            Thailand(categories=(ARMED_FORCES,), years=range(2006, 2008)),
+            ("2006-01-25", name),
+            ("2007-01-18", name),
         )
-        self.assertHolidayName(name, dt)
-        self.assertNoHolidayName(name, 1956, 1999)
 
-        self.assertHolidayName(name, (f"{year}-05-13" for year in range(1957, 1997)))
+    def test_bank_holiday(self):
+        a_name = "วันหยุดเพิ่มเติมสำหรับการปิดบัญชีประจำปีของธนาคารเพื่อการเกษตรและสหกรณ์การเกษตร"
+        m_name = "วันหยุดภาคครึ่งปีของสถาบันการเงินและสถาบันการเงินเฉพาะกิจ"
 
-        # No Royal Ploughing Ceremony on weekend for 1997-2023
+        # Start
+        self.assertHolidays(
+            Thailand(categories=(BANK,), years=range(1942, 1944)),
+            ("1943-04-01", a_name),
+            ("1943-07-01", m_name),
+        )
+        # End
+        self.assertHolidays(
+            Thailand(categories=(BANK,), years=range(2017, 2023)),
+            ("2017-04-01", a_name),
+            ("2017-07-01", m_name),
+            ("2018-04-01", a_name),
+            ("2018-07-01", m_name),
+            ("2019-04-01", a_name),
+            ("2020-04-01", a_name),
+            ("2021-04-01", a_name),
+        )
+
+    def test_school_holiday(self):
+        self.assertHolidays(
+            Thailand(categories=(SCHOOL,), years=range(1956, 1958)),
+            ("1957-01-16", "วันครู"),
+        )
+
```

Comment:
Need to test 1947 also, for full test coverage.

Reply:
Fixed.